instrumentation/log4j/log4j-appender-2.17/javaagent/build.gradle.kts

```diff
@@ -21,16 +21,33 @@ dependencies {
 
   testImplementation("org.awaitility:awaitility")
 
+  // this dependency is needed for the slf4j->log4j test
+  if (findProperty("testLatestDeps") as Boolean) {
+    testImplementation("org.apache.logging.log4j:log4j-slf4j2-impl:2.19.0")
+  } else {
+    // log4j 2.17 doesn't have an slf4j2 bridge
+    testImplementation("org.apache.logging.log4j:log4j-slf4j-impl:2.17.0")
+    testImplementation("org.slf4j:slf4j-api") {
+      version {
+        strictly("1.7.36")
+      }
+    }
+  }
+
   // this is needed for the async logging test
   testImplementation("com.lmax:disruptor:3.4.2")
 }
 
+tasks.withType<Test>().configureEach {
+  systemProperty("testLatestDeps", findProperty("testLatestDeps") as Boolean)
```

Comment:
Nit: you could extract `findProperty("testLatestDeps") as Boolean` to a variable